src/backends/perf_pt/mod.rs

```diff
@@ -223,15 +205,22 @@ impl Tracer for PerfPTTracer {
         Ok(())
     }
 
-    fn stop_tracing(&mut self) -> Result<(), TraceMeError> {
+    fn stop_tracing(&mut self) -> Result<HWTrace, TraceMeError> {
         let tr_ctx = self.tracer_ctx.ok_or(TraceMeError::TracerNotStarted)?;
+        let buf = ptr::null() as *const u8;
+        let len = 0;
         let rc = unsafe {
-            perf_pt_stop_tracer(tr_ctx)
+            perf_pt_stop_tracer(tr_ctx, &buf, &len)
         };
         if rc == -1 {
             return Err(TraceMeError::CFailure);
         }
-        Ok(())
+        assert!(buf != ptr::null()); // The C code should have put the buffer address here.
+        let trace = HWTrace::from_buf(buf, len);
+        if env::var("HWTRACE_DEBUG_DUMP_TRACES").is_ok() {
```

Comment:
This is quite an expensive call. Do we really need / want this functionality? I'm really not keen on the env vars trick: I'd prefer to have an extra function parameters or similar to control this sort of thing.

Reply:
I'm open to new ideas, but even if you have extra args, what triggers the debugging code? There needs to be an external interface.

Reply:
The interpreter author, IMHO, should be able to turn this on and off (if they really want to; although I can't imagine any reason for a normal interpreter author to do this). My problem with env vars as "global triggers" is that they're hard to discover, and users can occasionally trigger things that they never intended -- and may not even have realised that they've triggered something.

Reply:
OK. Well I guess that moves the problem up to the interpreter at least. They'd still need switches or environment vars, right?

I'm OK with that if you are.

Reply:
Yes, they'd be in charge of turning this on or off (however they want to).